api/envoy/config/core/v3/base.proto

```diff
@@ -87,6 +87,9 @@ message Locality {
   // into smaller chunks of sub-zones so they can be load balanced
   // independently.
   string sub_zone = 3;
+
+  // Metadata to provide additional information about the locality.
```

Comment:
@tonya11en can you remind me why this is preferred to doing it in `LocalityLbEndpoints` as a top-level shared field there?

As @wbpcode points out, all things being equal there would be a preference for a more local change if it's only used for endpoint transport socket match.

Reply:
There's no reason it has to be part of the Locality message as opposed to the LocalityLbEndpoints. I just thought it made sense to put it in Locality at the time of writing the patch.

I'll pull it up one level to LocalityLbEndpoints if that seems more appropriate to you.

Reply:
Yep, better, thanks.